Parser/Node.cs

```diff
@@ -75,6 +75,13 @@ public record NIfStmt (NExpr Condition, NStmt IfPart, NStmt? ElsePart) : NStmt {
    public override void Accept (Visitor visitor) => visitor.Visit (this);
 }
 
+// A Break statement
+// Break Token is needed just to throw error in case of exception
+public record NBreakStmt (Token Break, Token? Level) : NStmt {

```

Comment:
A lot of code will simplify if NBreakStmt is declared like this:
`public record NBreakStmt (Token Break, int Level) : NStmt {`
Then, you can avoid the Level == null in several places.